### PR TITLE
Settings: cautiously warn instead of erroring out for now

### PR DESCRIPTION
**Pull request by dmitryduev**
Description
-----------
Context: the new Settings object (https://github.com/wandb/client/pull/3083) makes the way we handle settings significantly more consistent, but also strict, which may potentially cause unforeseen problems for our users. Thus instead of erroring out everywhere, we:
- Error out only if that was what was happening before, with the old Settings.
- Display warnings if any other check is not successful (such as if a validator failed or a non-expected argument was passed to `wandb.Settings`).
- Store information about warnings in the Settings object and report that in telemetry.

image

Testing
-------
Added relevant tests to `wandb_run_test` and `wandb_settings_test`.

### CODE REVIEW

File: wandb/sdk/wandb_settings.py [hunk 1]
```diff
@@ -87,7 +87,11 @@ def _str_as_bool(val: Union[str, bool, None]) -> bool:
         return ret_val
     except (AttributeError, ValueError):
         pass
-    raise UsageError(f"Could not parse value {val} as a bool")
+    except Exception:
```
Comment (raubitsj) [inline wandb/sdk/wandb_settings.py:90]: is this right, or should the warn just be in the fallthrough case?

File: wandb/sdk/wandb_settings.py [hunk 2]
```diff
@@ -241,7 +263,17 @@ def _validate(self, value: Any) -> Any:
...
+                    #  remove this once we are confident
+                    if self.name in self.__strict_validate_settings:
+                        raise ValueError(
+                            f"Invalid value for property {self.name}: {value}"
+                        )
+                    else:
+                        wandb.termwarn(
+                            f"Invalid value for property {self.name}: {value}. "
+                            "This will cause an error in the future."
+                        )
```
Comment (raubitsj) [inline wandb/sdk/wandb_settings.py:276]: should you continue running more validators or should you break/return?

File: wandb/sdk/wandb_settings.py (hunk 1 above)
Comment (dmitryduev) [reply wandb/sdk/wandb_settings.py:90]: yea, fixed.

File: wandb/sdk/wandb_settings.py (hunk 2 above)
Comment (dmitryduev) [reply wandb/sdk/wandb_settings.py:276]: right, breaking now.

**Review by raubitsj — APPROVED** [review]

**Pull request merged**